NebulaNetwork/PacketProcessors/Planet/FactoryLoadRequestProcessor.cs

```diff
@@ -18,10 +20,19 @@ public override void ProcessPacket(FactoryLoadRequest packet, NebulaConnection c
 
             using (BinaryUtils.Writer writer = new BinaryUtils.Writer())
             {
-                factory.Export(writer.BinaryWriter);
+                if (Config.Options.NetworkSavegameCompression)
```

Comment:
The config option should only be used for the host. The host should send their option to the client when the client connects, and the client should determine which system to expect based off of that.

Reply:
There is no need to communicate the setting to the client, because the determination what decoding to use made based on the received data (you can view this in the patching of the CargoPath_Patch.cs Import function). This ensures that decoding also works for data that is stored in a normal savegame.